src/cms/views/push_notifications/push_notification_sender.py

```diff
@@ -1,43 +1,133 @@
+    """
+
     logger = logging.getLogger(__name__)
     fcm_url = "https://fcm.googleapis.com/fcm/send"
 
-    """
-    Sends push notifications via FCM legacy http api.
-    Returns boolean indicating success or failure
-    """
+    def __init__(self, push_notification):
+        """
+        Load relevant push notification translations and prepare content for sending
 
-    # pylint: disable=too-many-arguments
-    def send(self, region_slug, channel, title, message, lan_code):
+        :param push_notification: the push notification that should be sent
+        :type push_notification: ~cms.models.push_notifications.push_notification.PushNotification
+        """
+        self.push_notification = push_notification
+        self.prepared_pnts = []
+        self.primary_pnt = PushNotificationTranslation.objects.get(
+            push_notification=push_notification,
+            language=push_notification.region.default_language,
+        )
+        if len(self.primary_pnt.title) > 0:
+            self.prepared_pnts.append(self.primary_pnt)
+        self.load_secondary_pnts()
+        self.auth_key = self.get_auth_key()
+
+    def load_secondary_pnts(self):
+        """
+        Load push notification translations in other languages
+        """
+        secondary_pnts = PushNotificationTranslation.objects.filter(
+            push_notification=self.push_notification
+        ).exclude(id=self.primary_pnt.id)
+        for secondary_pnt in secondary_pnts:
+            if (
+                secondary_pnt.title == ""
+                and pnt_const.USE_MAIN_LANGUAGE == self.push_notification.mode
+            ):
+                secondary_pnt.title = self.primary_pnt.title
+                secondary_pnt.text = self.primary_pnt.text
+                self.prepared_pnts.append(secondary_pnt)
+            if len(secondary_pnt.title) > 0:
+                self.prepared_pnts.append(secondary_pnt)
+
+    def is_valid(self):
+        """
+        Check if all data for sending push notifications is available
+
+        :return: all prepared push notification translations are valid
+        :rtype: bool
+        """
+        if self.auth_key is None:
+            return False
+        for pnt in self.prepared_pnts:
+            if not pnt.title:
+                self.logger.info("Push Notification Translation invalid: %s", str(pnt))
+                return False
+        return True
+
+    def get_auth_key(self):
+        """
+        Get FCM API auth key
+
+        :return: FCM API auth key
+        :rtype: str
+        """
         fcm_auth_config_key = "fcm_auth_key"
-        auth_key = Configuration.objects.filter(key=fcm_auth_config_key)
-        if auth_key.exists():
+        auth_key = Configuration.objects.filter(key=fcm_auth_config_key).first().value
+        if auth_key:
             self.logger.info("Got fcm_auth_key from database")
+            return auth_key
```

Comment:
If the auth key does not exist in the database, this will result in the error: `'NoneType' object has no attribute 'value'`.
So I suggest to keep the previous if-condition.
```suggestion
        auth_key = Configuration.objects.filter(key=fcm_auth_config_key)
        if auth_key.exists():
            self.logger.info("Got fcm_auth_key from database")
            return auth_key.first().value
```

Reply:
Done.